unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/CharacterPreview/PreviewCameraPanning.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using Vector3 = UnityEngine.Vector3;
+
+namespace MainScripts.DCL.Controllers.HUD.CharacterPreview
+{
+    public class PreviewCameraPanning : MonoBehaviour, IDragHandler, IBeginDragHandler, IEndDragHandler
```

Comment:
I guess this requires a Monobehaviour to get the drag handlers. But we can use a Humble Object pattern, and have a POCO and inject a smaller monobehaviour wrapping the drag events.

Reply:
Following your suggested approach, correct me if I'm wrong, we would have:
- A `CameraPanningDetector` MonoBehaviour class that would be almost exactly the same as now.
- A `PreviewCameraPanningController` POCO class that would expose a `OnPanning` event (to propagate the `OnPanning` from the MonoBehaviour) and would play the begin/end drag sounds.

All the logic of the current class (except the triggering of 2 sounds) depends directly from the `MonoBehaviour` flow... do you think it worth it to apply that separation? What would be the bebefit?

Reply:
Refactor done!

I finally created:
- `CharacterPreviewInputDetector` : `ICharacterPreviewInputDetector` -> This MonoBehaviour class controls the minimum needed about mouse inputs detection (`IBeginDragHandler, IDragHandler, IEndDragHandler, IPointerEnterHandler, IPointerExitHandler`).
- `PreviewCameraPanningController` (POCO)
- `PreviewCameraZoomController` (POCO)
- `PreviewCameraRotationController` (POCO)

**PreviewCameraPanning refactor:** https://github.com/decentraland/unity-renderer/pull/5162/commits/114cfabdceb095a990400458828e9463b7982390
**PreviewCameraZoom & PreviewCameraRotation refactor:** https://github.com/decentraland/unity-renderer/pull/5162/commits/42c3f5e138c156bd6476896a049c0e5b9b73297c